File: Assessment/Assessment.WEB/src/app/services/assessment.service.ts
import { Inject } from '@angular/core';
import { Http } from '@angular/http';
import { Assessment } from '../entities/assessment';

export class AssessmentService {
    http: Http;
    baseUrl: string;
    controller: string = "api/Assessment/";

    constructor(http: Http, baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
    }

    public getAll() {
        return this.http.get(this.baseUrl + this.controller);
    }

    public get(id: number) {
        return this.http.get(this.baseUrl + this.controller + id);
    }

    public getQuestions(id: number) {
        return this.http.get(this.baseUrl + this.controller + "Questions/" + id);
    }

    public update(assessment: Assessment) {
        return this.http.put(this.baseUrl + this.controller, assessment);
    }

    public insert(assessment: Assessment) {
        return this.http.post(this.baseUrl + this.controller, assessment);
    }

    public delete(id: number) {
        return this.http.delete(this.baseUrl + this.controller + id);
    }
}

File: Assessment/Assessment.Entities/AssessmentService.cs
using System;
using System.Collections.Generic;
using Assessment.Entities.DataInterface;

namespace Assessment.Entities
{
    public class AssessmentService
    {
        private IAssessmentRepository _repository;

        public AssessmentService(IAssessmentRepository repository)
        {
            _repository = repository;
        }

        public void Insert(Assessment assessment)
        {
            _repository.Insert(assessment);
        }

        public void Delete(int assessmentID)
        {
            _repository.Delete(assessmentID);
        }

        public void Update(Assessment assessment)
        {
            _repository.Update(assessment);
        }

        public Assessment Get(int assessmentID)
        {
            return _repository.Get(assessmentID);
        }

        public IEnumerable<Assessment> GetAll()
        {
            return _repository.GetAll();
        }

        public IEnumerable<Question> GetQuestions(int assessmentID)
        {
            return _repository.GetQuestions(assessmentID);
        }

        public IEnumerable<Answer> GetAnswers(int assessmentID)
        {
            return _repository.GetAnswers(assessmentID);
        }
    }
}